### PR TITLE
Some tidying-up and some new funcitonality

### PR DESCRIPTION
**Pull request by jimpil**
Hi there,

First of all many thanks for your work on this ;)
Here is a PR that does a major cleanup, gets rid of reflection (there were more than 50 reflective calls!), swaps `cheshire` for `clojure.data.json` (also removes wallhack), and adds a new `ids.clj` namespace for dealing with ID-generation (pretty basic stuff).

Let me know if you're interested :) 

Kind regards,

### CODE REVIEW

File: src/chazel/core.clj [hunk 1]
```diff
@@ -21,231 +20,254 @@
...
+            local-update-policy  (.setLocalUpdatePolicy local-update-policy)
+            cache-local-entries  (.setCacheLocalEntries cache-local-entries))))
 
 (defn with-creds
   ([creds]
    (with-creds creds (Config.)))
   ([{:keys [group-name group-password]} config]
-   (.setGroupConfig config (GroupConfig. group-name
-                                         group-password))))
+   (println group-name group-password )
```
Comment (tolitius) [inline src/chazel/core.clj]: let's not _print_ creds

**Review by tolitius — CHANGES_REQUESTED** [review]

**Comment by tolitius** [comment]
:metal: thanks a lot. this is _a lot of_ cleanup )
did you repl / test the functions you modified?

File: src/chazel/core.clj (hunk 1 above)
Comment (jimpil) [reply src/chazel/core.clj]: oops!

**Comment by jimpil** [comment]
Hi there,

I've run most of the code shown in the README and I can confirm it works. If you happen to have 5-10 minutes to spend using it in a real project - that would be awesome :).

**Comment by tolitius** [comment]
I unfortunately do not have a prod like hazelcast env at the moment.
I did do some smoke testing. And since you have not really changed the _features_ besides the `IdGenerator` (thank you) it looks good.

could you add some documentation about the `IdGenerator` so other people could find it easier?

I will merge this for now, since this code is written over the years and did collect some tech dept that you sliced and diced, hence it is definitely better than it was.

**Comment by tolitius** [comment]
it now lives in `[chazel "0.1.20"]`

**Pull request merged**